### PR TITLE
Implemented linear interpolation for FloatBox2D.

### PR DESCRIPTION
**Pull request by davidpiuva**
Interpolating FloatBox2D.

### CODE REVIEW

File: source/math/FloatBox2D.ooc [hunk 1]
```diff
@@ -187,4 +187,12 @@ FloatBox2D: cover {
 		}
 		This new(xMinimum, yMinimum, xMaximum - xMinimum, yMaximum - yMinimum)
 	}
+	linearInterpolation: static func (a, b: This, ratio: Float) -> This {
+		FloatBox2D new(
```
Comment (marcusnaslund) [inline source/math/FloatBox2D.ooc]: as magic says, `This new`

File: source/math/FloatBox2D.ooc [hunk 2]
```diff
@@ -187,4 +187,12 @@ FloatBox2D: cover {
...
+			Float linearInterpolation(a left, b left, ratio),
+			Float linearInterpolation(a top, b top, ratio),
+			Float linearInterpolation(a width, b width, ratio),
+			Float linearInterpolation(a height, b height, ratio)
+		)
```
Comment (marcusnaslund) [inline source/math/FloatBox2D.ooc]: I think I also prefer this parenthesis together with the one on the previous row. @davidhesselbom ?

Comment (davidpiuva) [inline source/math/FloatBox2D.ooc]: Then the other lines must have consistent indentation.

Comment (marcusnaslund) [inline source/math/FloatBox2D.ooc]: What do you mean? Magic says nothing.

Comment (davidhesselbom) [inline source/math/FloatBox2D.ooc]: I think you can put it all on one line, actually:
```ooc
FloatBox2D createFrom(FloatPoint2D linearInterpolation(a center,  b center), FloatSize2D linearInterpolation(a size, b size))
```

Comment (davidpiuva) [inline source/math/FloatBox2D.ooc]: When jumping more than one step to the left on the same line, it looks broken and draws attention.

Comment (davidpiuva) [inline source/math/FloatBox2D.ooc]: ok

Comment (davidhesselbom) [inline source/math/FloatBox2D.ooc]: There may be more efficient ways of computing it, though. Maybe 
```ooc
FloatBox2D createFrom(FloatPoint2D linearInterpolation(a leftTop,  b leftTop), FloatSize2D linearInterpolation(a size, b size))
```
works too, I'm a bit too tired to do the math right now. If you'd added a test for this function in this PR, it'd be easy to find out.

File: source/math/FloatBox2D.ooc [hunk 3]
```diff
@@ -187,4 +187,8 @@ FloatBox2D: cover {
 		}
 		This new(xMinimum, yMinimum, xMaximum - xMinimum, yMaximum - yMinimum)
 	}
+	linearInterpolation: static func (a, b: This, ratio: Float) -> This {
+		This new(Float linearInterpolation(a left, b left, ratio), Float linearInterpolation(a top, b top, ratio),
+		Float linearInterpolation(a width, b width, ratio), Float linearInterpolation(a height, b height, ratio))
```
Comment (davidhesselbom) [inline source/math/FloatBox2D.ooc]: I don't see how you could end up with this after the suggestion I gave you.

**Comment by davidhesselbom** [comment]
Please add a test.

**Comment by davidpiuva** [comment]
Added a test.

**Comment by marcusnaslund** [comment]
magic complains @davidpiuva

**Pull request merged**